### PR TITLE
Use towncrier and a custom tox environment to prepare a release

### LINKED ISSUE #66: adapting towncrier and release automation
we should take a look for handling this here as well

### PR DESCRIPTION
**Pull request by nicoddemus**
Mostly borrowed by pytest and tox. 😉 

Fix #66

### CODE REVIEW

File: changelog/_template.rst [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+{% if sections[section] %}
+{% for category, val in definitions.items() if category in sections[section] %}
+
+{{ definitions[category]['name'] }}
+{{ underline * definitions[category]['name']|length }}
+
+{% if definitions[category]['showcontent'] %}
+{% for text, values in sections[section][category]|dictsort(by='value') %}
+{% set issue_joiner = joiner(', ') %}
+- {% for value in values|sort %}{{ issue_joiner() }}`{{ value }} <https://github.com/pytest-dev/pytest/issues/{{ value[1:] }}>`_{% endfor %}: {{ text }}
```
Comment (tgoodlet) [inline changelog/_template.rst]: @nicoddemus is url link here correct?

**Comment by tgoodlet** [comment]
@nicoddemus should I make a separate PR then adding all the `newsfragments` for what would have been the last set of changes leading up to the next minor release?

File: changelog/_template.rst (hunk 1 above)
Comment (nicoddemus) [reply changelog/_template.rst]: Oh definitely, good catch!

**Comment by nicoddemus** [comment]
Fixed the URL, good catch!

**Comment by nicoddemus** [comment]
>  should I make a separate PR then adding all the newsfragments for what would have been the last set of changes leading up to the next minor release?

Sure, that would be great, thanks! :grin:

This will make the next releases much simpler now. 👍

**Pull request merged**